reference/reference.go

```diff
@@ -195,40 +204,139 @@ func ParseNamed(s string) (Named, error) {
 	return named, nil
 }
 
-// WithName returns a named object representing the given string. If the input
-// is invalid ErrReferenceInvalidFormat will be returned.
-func WithName(name string) (Named, error) {
```

Comment:
Could this function be preserved, perhaps under the name `ParseNameOnly`? It is used in several places in the engine.